sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
@viirya and @cloud-fan, How about adding few notes that implies `columnNameOfCorruptRecord` for both CSV and JSON is currently incomplete like union type in Hive for now?

I can't think of a better way than parsing / converting everything always but I guess of course this is not what we want for now. I also am aware of few such issues about this.